scripts/test_zmq/test_zmq/server.py

```diff
@@ -0,0 +1,101 @@
+#! /usr/bin/env python3
+
+import argparse
+import shutil
+import socket
+from concurrent.futures import ThreadPoolExecutor
+from tempfile import TemporaryDirectory
+from typing import NamedTuple
+
+import zmq
+from test_zmq.zstack import ZStack
+
+
+SEED = b'E21bEA7DeaE981cBabCECd9FAeF4e340'
+EXPECTED_ZMQ_REPLY = '{"type": "DIAGNOSE", "text": "ZMQ connection is possible"}'
+EXPECTED_TCP_REPLY = '{"type": "DIAGNOSE", "text": "TCP connection is possible"}'
+QUIT = False
+
+HA = NamedTuple("HA", [
+    ("host", str),
+    ("port", int)])
+
+
+def msg_handler(zstack, msg):
+    _, frm = msg
+    print(msg)
+    zstack.send(EXPECTED_ZMQ_REPLY, frm)
+
+
+def loop():
+    loop = zmq.asyncio.ZMQEventLoop()
+    return loop
+
+
+class SafeTemporaryDirectory(TemporaryDirectory):
+
+    @classmethod
+    def _cleanup(cls, name, warn_message):
+        shutil.rmtree(name, ignore_errors=True)
+
+    def cleanup(self):
+        if self._finalizer.detach():
+            shutil.rmtree(self.name, ignore_errors=True)
+
+
+def up_zmq_server(server_ha):
+    print("ZMQ_SERVER: Seed is {}".format(SEED))
+    server = ZStack(name='Test_zmq',
+                    ha=server_ha,
+                    basedirpath=base_dir,
+                    msgHandler=msg_handler,
+                    seed=SEED,
+                    onlyListener=True)
+    server.start()
+    return server
+
+
+def up_tcp_server(server_ha):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(server_ha)
+        s.listen()
+        print("TCP_SERVER: Listen clients on {}".format(server_ha))
+        while True and not QUIT:
```

Comment:
It looks like QUIT is always False, isn't it?

Reply:
I forgot to remove it. It was some experiments need for demo.